Make meteor game mode great again

optimumtact opened the pull request:
Reverts tgstation/-tg-station#13907

Thanks to the new lag removal, we can make meteors great again.

PKPenguin321 commented:
rip lag removal

optimumtact commented:
Maybe we outta test merge it first then

Zilenan91 commented:
set it to 9 this couldn't possibly go wrong

GunHog commented:
I like it. Make sure you **TEST** this locally, first.

optimumtact commented:
There's no point testing this locally because I can't stress a box anywhere near what a proper test on Sybil/Basil would do.

Dorsisdwarf commented:
Test-merge would work for live testing

optimumtact commented:
hg test merged this about 8 hour ago

MrStonedOne commented:
Make it ramp up to 6 over the course of X amount of time. like over 30 minutes.

optimumtact commented:
:no_good: :rotating_light: :lock:

Dorsisdwarf commented:
PR no good,  activate alarms and lock at once?

MrsTonedOne committed:
Makes meteor math great again.

MrStonedOne commented:
https://github.com/optimumtact/-tg-station/pull/6

optimumtact commented:
:ticket: :green_heart: :+1:

hornygranny commented:
the test merge failed when i did it actually someone test merge this

MrsTonedOne committed:
Fixes some meteor mode bugs

duncathan commented on code/game/gamemodes/meteor/meteor.dm:
```diff
@@ -11,17 +12,20 @@
+	if (prob(meteorminutes/2))
+		wavetype = meteors_catastrophic
+	else if (prob(meteorminutes))
+		wavetype = meteors_threatening
```
Switch the order of these two, and remove the `else`. It'll give you more accurate results as `meteorminutes` increases.

optimumtact replied:
https://file.house/tQG9.png

duncathan replied:
Yes that's what removing the else is for
The else causes that issue as well as slightly inaccurate probabilities

optimumtact replied:
I shall make it so then

optimumtact committed:
Minor changes for Dunc plus readability improvements

optimumtact commented:
what the fuck

PKPenguin321 commented:
?

optimumtact commented:
Accidentally force pushed the wrong branch over this

@duncathan did what you said

I also rewrote the branch to fix up the commit history and cleaned things up a bit for readability.
